src/QueryResults.ts

```diff
@@ -0,0 +1,62 @@
+import Promise, { Thenable, Executor } from 'dojo-core/Promise';
+import Task from 'dojo-core/async/Task';
+import {duplicate, mixin} from 'dojo-core/lang';
+import * as dstore from './interfaces';
+
+export interface QueryOptions {
+	totalLength?: number| Promise<number>;
+	response: any;
+	length?: number;
+	forEach?: <T>(callback:(value: T, index: number, source: T[]) => any, instance: any) => Promise<void>;
+}
+
+function forEach<T>(callback: (value: T, index?: number, array?: T[]) => void, thisObject?: any): Promise<void> {
+	const execute: Executor<T[]> = function (resolve, reject) {
```

Comment:
I'd vouch for a `TODO` comment here indicating that it might be worth just simplifying to use `Promise.resolve` (analogous to single-argument `when`).  I've made a note on #137 for the future about simplifying `QueryResults` in general to better account for the fact that dstore stores now always expose async APIs.